packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -756,24 +792,21 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         channel.handle.attachGraph();
 
         assert(this.isAttached, 0x182 /* "Data store should be attached to attach the channel." */);
-        // Get the object snapshot only if the data store is Bound and its graph is attached too,
-        // because if the graph is attaching, then it would get included in the data store snapshot.
-        if (this.bindState === BindState.Bound && this.graphAttachState === AttachState.Attached) {
```

Comment:
Is making it unconditional a result of realizing that condition was always true, or changes in the flow where we never get here when condition was false in the past?